src/htm/types/Exception.hpp

```diff
@@ -158,6 +163,8 @@ class Exception : public std::runtime_error { //TODO rename to NTAException to b
   UInt32 lineno_;
   mutable std::string message_; //mutable bcs modified in getMessage which is used in what() but that needs be const
   std::string stackTrace_;
+  mutable std::string what_;
```

Comment:
Yes.  The what() function is an override of the std::exception so we cannot change its signature. It returns a char*.  This must point to something that will have a scope that last at least as long as the Exception class instance.  If it is a local variable it goes away before it can be returned.  So that is why I made it a class member.